pkg/packagekit/updates.jsx

```diff
@@ -513,116 +513,106 @@ class RestartServices extends React.Component {
-                        });
-            },
-        });
-    }
+                    // scroll update log to the bottom, if it already is (almost) at the bottom
+                    if (log && atBottom)
+                        log.scrollTop = log.scrollHeight;
+                });
+    };
 
-    componentWillUnmount() {
-        this._mounted = false;
+    useEffect(() => {
+        // read the current package from the transaction
+        updateForProperties(null, null);
+        // keep updating it when new Package signals come in
+        PK.watchTransaction(transaction, {
+            Package: (status, packageId) => updateForProperties(status, packageId),
+        });
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+    const cancelButton = (
+        <Button className={actions.length !== 0 && "progress-cancel"}
+                variant="secondary"
+                onClick={onCancel}
+                isDisabled={!allowCancel}>
+            {_("Cancel")}
+        </Button>
+    );
+    if (actions.length === 0) {
+        return <EmptyStatePanel title={ _("Initializing...") }
+                                headingLevel="h5"
+                                titleSize="4xl"
+                                secondary={cancelButton}
+                                loading
+        />;
     }
 
-    render() {
-        const cancelButton = (
-            <Button className={this.state.actions.length !== 0 && "progress-cancel"}
-                   variant="secondary"
-                   onClick={this.props.onCancel}
-                   isDisabled={!this.props.allowCancel}>
-                {_("Cancel")}
-            </Button>
-        );
-        if (this.state.actions.length === 0) {
-            return <EmptyStatePanel title={ _("Initializing...") }
-                                    headingLevel="h5"
-                                    titleSize="4xl"
-                                    secondary={cancelButton}
-                                    loading
-            />;
-        }
-
-        const lastAction = this.state.actions[this.state.actions.length - 1];
-        const timeRemaining = this.state.timeRemaining && timeformat.distanceToNow(new Date().valueOf() + this.state.timeRemaining * 1000);
-        return (
-            <>
-                <div className="progress-main-view">
-                    <div className="progress-description">
-                        <Spinner isSVG size="sm" />
-                        <strong>{ PK_STATUS_STRINGS[lastAction.status] || PK_STATUS_STRINGS[PK.Enum.STATUS_UPDATE] }</strong>
-                        &nbsp;{lastAction.package}
-                    </div>
-                    <Progress title={timeRemaining} value={this.state.percentage} />
-                    {cancelButton}
+    const lastAction = actions[actions.length - 1];
+    const formatTimeRemaining = timeRemaining && timeformat.distanceToNow(new Date().valueOf() + timeRemaining * 1000);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17517-20220705-092548-cbb7d34f-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)